rust/arrow/src/array/equal.rs

```diff
@@ -1046,6 +1062,30 @@ impl PartialEq<FixedSizeBinaryArray> for Value {
     }
 }
 
+impl JsonEqual for UnionArray {
+    fn equals_json(&self, _json: &[&Value]) -> bool {
+        unimplemented!()
```

Comment:
Is this intentionally unimplemented? If so, would be good to add a comment explaining that.